composeApp/src/androidMain/kotlin/sp/bvantur/tasky/core/data/TaskySyncScheduler.android.kt

```diff
@@ -3,41 +3,84 @@ package sp.bvantur.tasky.core.data
 import androidx.work.BackoffPolicy
 import androidx.work.Constraints
 import androidx.work.Data
+import androidx.work.ExistingWorkPolicy
 import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequest
 import androidx.work.OneTimeWorkRequestBuilder
 import androidx.work.WorkManager
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
+import sp.bvantur.tasky.core.data.local.EventEntity
+import sp.bvantur.tasky.core.data.local.SyncStep
 import java.util.concurrent.TimeUnit
 
 actual class TaskySyncScheduler(
     private val workManager: WorkManager,
     private val applicationCoroutineScope: CoroutineScope
 ) {
-    actual suspend fun scheduleAgendaSync(eventIds: List<String>) {
-        eventIds.forEach { eventId ->
-            val agendaWorker = OneTimeWorkRequestBuilder<SyncEventsWorker>()
-                .setConstraints(
-                    Constraints.Builder()
-                        .setRequiredNetworkType(NetworkType.CONNECTED)
-                        .build()
-                )
-                .setBackoffCriteria(
-                    backoffPolicy = BackoffPolicy.EXPONENTIAL,
-                    backoffDelay = 5000,
-                    timeUnit = TimeUnit.MILLISECONDS
-                )
-                .addTag("add_event")
-                .setInputData(
-                    Data.Builder()
-                        .putString(SyncEventsWorker.SYNC_EVENT_ID, eventId)
-                        .build()
-                )
-                .build()
+    actual suspend fun scheduleAgendaSync(events: List<EventEntity>) {
+        events.forEach { event ->
+            val agendaWorker: OneTimeWorkRequest = when (event.syncStep) {
+                SyncStep.CREATE -> {
+                    onCreateEventWorker(event)
+                }
+                SyncStep.EDIT -> {
+                    return@forEach // TODO
+                }
+                SyncStep.DELETE -> {
+                    onDeleteEventWorker(event)
+                }
+
+                else -> return@forEach
+            }
```

Comment:
Have you considered the scenario where a user is offline, creates an event, and then immediately updates the event? Would your approach here override the CREATE sync step to an EDIT one? Your remote api call would fail the update call since the event ID would not exist on the backend.

Reply:
I didn't think about this use case yet. I will add TODO just to not forget about this use case. Thanks!